pyintegration/tests/test_table_factory.py

```diff
@@ -100,11 +109,71 @@ def test_input_column_error(self):
         with self.assertRaises(DHError):
             bool_col(name="Boolean", data=[True, j_al])
 
+    def test_dynamic_table_writer(self):
+        with self.subTest("Correct Input"):
+            col_defs = {"Numbers": dtypes.int32, "Words": dtypes.string}
+            with DynamicTableWriter(col_defs) as table_writer:
+                table_writer.write_row(1, "Testing")
+                table_writer.write_row(2, "Dynamic")
+                table_writer.write_row(3, "Table")
+                table_writer.write_row(4, "Writer")
+                result = table_writer.table
+                self.assertTrue(result.is_refreshing)
+
+        with self.subTest("One too many values in the arguments"):
+            with DynamicTableWriter(col_defs) as table_writer, self.assertRaises(DHError) as cm:
+                table_writer.write_row(1, "Testing", "shouldn't be here")
+            self.assertIn("RuntimeError", cm.exception.root_cause)
+
+        with self.subTest("Proper numerical value conversion"):
+            col_defs = {
+                "Double": dtypes.double,
+                "Float": dtypes.float32,
+                "Long": dtypes.long,
+                "Int32": dtypes.int32,
+                "Short": dtypes.short,
+                "Byte": dtypes.byte
+            }
+            with DynamicTableWriter(col_defs) as table_writer:
+                table_writer.write_row(10, 10, 10, 10, 10, 10)
+                table_writer.write_row(10.1, 10.1, 10.1, 10.1, 10.1, 10.1)
+
+        with self.subTest("Incorrect value types"):
+            with DynamicTableWriter(col_defs) as table_writer, self.assertRaises(DHError) as cm:
+                table_writer.write_row(10, '10', 10, 10, 10, '10')
+            self.assertIn("RuntimeError", cm.exception.root_cause)
+
+    def test_historical_tale_replayer(self):
+        dt1 = to_datetime("2000-01-01T00:00:01 NY")
+        dt2 = to_datetime("2000-01-01T00:00:03 NY")
+        dt3 = to_datetime("2000-01-01T00:00:06 NY")
+
+        hist_table = new_table([
+            datetime_col("DateTime", [dt1, dt2, dt3]),
+            int_col("Number", [1, 3, 6])]
+        )
+
+        start_time = to_datetime("2000-01-01T00:00:00 NY")
+        end_time = to_datetime("2000-01-01T00:00:07 NY")
+
+        replayer = TableReplayer(start_time, end_time)
+        replay_table = replayer.add_table(hist_table, "DateTime")
+        replay_table2 = replayer.add_table(hist_table, "DateTime")
+        self.assertEqual(replay_table, replay_table2)
+
+        replayer.start()
+        self.assertTrue(replay_table.is_refreshing)
+        self.assertTrue(replay_table2.is_refreshing)
+        sleep(1)
+        replayer.shutdown()
 
-@dataclass
-class CustomClass:
-    f1: int
-    f2: str
+        with self.assertRaises(DHError) as cm:
+            replay_table3 = replayer.add_table(hist_table, "DateTime")
+        self.assertIn("RuntimeError", cm.exception.root_cause)
```

Comment:
Looks like you are expecting a failure if you register after starting.  This should be reflected in the javadoc above.